### PR TITLE
services: add support for filtered read_all

### LINKED ISSUE #38: Override read_all to filter by type like search
invenio-vocabularies's overrides its search method to filter by type:

def search(self, identity, params=None, es_preference=None, type=None,  # <-- this
           **kwargs)

. A similar `read_all` is needed in `VocabulariesService` to only read all a specific vocabulary type documents. 

In particular, this is needed if we want to use `read_all` to get all possible resource types in the deposit page dropdown.

### PR DESCRIPTION
**Pull request by ppanero**
requires https://github.com/inveniosoftware/invenio-records-resources/pull/250
closes #38 

- Note there is a parameter order change in the service methods to make the vocabulary type mandatory
- No tests are added since the method was already tested by the `test_facets.py` (not a valid excuse, but is good for now)

### CODE REVIEW

**Comment by ppanero** [comment]
Once merged needs to be migrated here https://github.com/inveniosoftware/invenio-app-rdm/pull/853/files#r638535062 too

File: invenio_vocabularies/services/service.py [hunk 1]
```diff
@@ -105,7 +107,7 @@ def create_type(self, identity, id, pid_type):
         db.session.commit()
         return type_
 
-    def search(self, identity, params=None, es_preference=None, type=None,
+    def search(self, identity, type, params=None, es_preference=None,
```
Comment (lnielsen) [inline invenio_vocabularies/services/service.py:110]: This will break with this change https://github.com/inveniosoftware/invenio-vocabularies/blob/master/invenio_vocabularies/resources/resource.py#L106

Now the question is why the tests don't break 😄 

To avoid problems before release, please revert this change

File: invenio_vocabularies/services/service.py [hunk 2]
```diff
@@ -135,3 +137,35 @@ def search(self, identity, params=None, es_preference=None, type=None,
...
         )
+
+    def read_all(self, identity, fields, type, cache=True, **kwargs):
+        """Search for records matching the querystring."""
+        cache_key = "-".join(fields)
+        results = current_cache.get(cache_key)
+        es_query = Q("match_all")
+
+        # If not found, NoResultFound is raised (caught by the resource).
+        vocabulary_type = VocabularyType.query.filter_by(id=type).one()
```
Comment (lnielsen) [inline invenio_vocabularies/services/service.py:148]: This will perform a database query and should be avoided when the results are cached.

File: invenio_vocabularies/services/service.py (hunk 1 above)
Comment (ppanero) [reply invenio_vocabularies/services/service.py:110]: Tests do not break because the call is made with "kwargs" i.e. type=... not just assuming the "position". But yes, ill revert.

File: invenio_vocabularies/services/service.py (hunk 2 above)
Comment (ppanero) [reply invenio_vocabularies/services/service.py:148]: we wont be able to reconstruct the search fully then... but since there are no links it should be potentially fine...
